modules/gdscript/doc_classes/@GDScript.xml

```diff
@@ -735,6 +735,34 @@
 				[b]Note:[/b] As annotations describe their subject, the [annotation @tool] annotation must be placed before the class definition and inheritance.
 			</description>
 		</annotation>
+		<annotation name="@tool_button">
+			<return type="void" />
+			<param index="0" name="text" type="String" default="&quot;&quot;" />
+			<param index="1" name="icon" type="StringName" default="&quot;&quot;" />
+			<description>
+				Mark a function to be displayed as a clickable button in the inspector when running the editor. The parameters are optional and can be used to customize the button.
+				If [param text] is specified it is used as the label text of the button. If [param text] is unspecified then the label text will be derived from the name of the annotated function and automatically capitalized.
+				If [param icon] is specified it is used to fetch a suitable icon texture for the button via [method Control.get_theme_icon] by selecting the desired [param icon] from the [code]"EditorIcons"[/code] theme type. If [param icon] is unspecified the default [code]"Callable"[/code] icon will be used instead.
+				The first parameter of the annotated function is also optional and is an [EditorUndoRedoManager]. Consider using the optionally passed [EditorUndoRedoManager] to allow the action to be reverted safely. The code snippet below demonstrates this:
+				[codeblock]
+				@tool
+				extends Sprite2D
+
+				@tool_button
+				func toot():
+				    print("toot!")
```

Comment:
Let's just make it "Thanks for reading!", very weird to make a comment like that in the docs 🙃

Reply:
'Toot' is literally the harmless onomatopoeia of a horn, or an alert, as per the `Callable` icon. How you feel about it isn't how it's read. I will curtail the personality of the documentation as follows, since it is traditional:

```gdscript
@tool_button
func hello():
    print("hello world!")
````